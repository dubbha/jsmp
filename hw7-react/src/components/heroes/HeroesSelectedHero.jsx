import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const HeroesSelectedHero = ({ hero }) => (
  <div className="selectedHero">
    <h2 className="selectedHero__header">
      <span className="uppercase">{hero.name}</span> is my hero
    </h2>
    <Link to={`/hero-details/${hero.id}`} className="selectedHero__link">View Details</Link>
  </div>
);

HeroesSelectedHero.propTypes = {
  hero: PropTypes.object.isRequired,
};

export default HeroesSelectedHero;
