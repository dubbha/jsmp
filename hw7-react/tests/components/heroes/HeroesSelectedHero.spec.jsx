import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HeroesSelectedHero from 'components/heroes/HeroesSelectedHero';

describe('HeroesSelectedHero', () => {
  it('should render selected hero', () => {
    const props = {
      hero: { id: 777, name: 'Zero' }
    };
    const wrapper = shallow(<HeroesSelectedHero {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
