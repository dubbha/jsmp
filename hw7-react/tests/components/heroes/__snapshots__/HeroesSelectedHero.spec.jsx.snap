// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroesSelectedHero should render selected hero 1`] = `
<div
  className="selectedHero"
>
  <h2
    className="selectedHero__header"
  >
    <span
      className="uppercase"
    >
      Zero
    </span>
     is my hero
  </h2>
  <Link
    className="selectedHero__link"
    replace={false}
    to="/hero-details/777"
  >
    View Details
  </Link>
</div>
`;
